src/nasem_dairy/NASEM_equations/micronutrient_requirement_equations.py

```diff
@@ -0,0 +1,889 @@
+    return An_Cu_y
+
+
+def calculate_An_Cu_l(Trg_MilkProd: float) -> float:
+    """
+    An_Cu_l: Cu required for lactation, mg/d
+    """
+    if Trg_MilkProd is None or math.isnan(Trg_MilkProd):    # Line 3054
+        An_Cu_l = 0
+    else:
+        An_Cu_l = 0.04 * Trg_MilkProd
+    return An_Cu_l
+
+
+def calculate_An_Cu_req(An_StatePhys: str, Dt_DMIn_ClfLiq: float, An_Cu_Clf: float, An_Cu_m: float, An_Cu_g: float, An_Cu_y: float, An_Cu_l: float) -> float:
+    """
+    An_Cu_req: Cu requirement, mg/d
+    """
+    if An_StatePhys == 'Calf' and Dt_DMIn_ClfLiq > 0:   # Line 3055-3056
+        An_Cu_req = An_Cu_Clf
+    else:
+        An_Cu_req = An_Cu_m + An_Cu_g + An_Cu_y + An_Cu_l
+    return An_Cu_req
+
+
+def calculate_An_Cu_bal(Abs_CuIn: float, An_Cu_req: float) -> float:
+    """
+    An_Cu_bal: Cu balance, mg/d
+    """
+    An_Cu_bal = Abs_CuIn - An_Cu_req    # Line 3057
+    return An_Cu_bal
+
+
+def calculate_An_Cu_prod(An_Cu_y: float, An_Cu_l: float, An_Cu_g: float) -> float:
+    """
+    An_Cu_prod: Cu for production?, mg/d
+    """
+    An_Cu_prod = An_Cu_y + An_Cu_l + An_Cu_g    # Line 3058
+    return An_Cu_prod
+
+
+### IODINE ###
+def calculate_An_I_req(An_StatePhys: str, An_DMIn: float, An_BW: float, Trg_MilkProd: float) -> float:
+    """
+    An_I_req: I requirement, mg/d
+    """
+    if An_StatePhys == 'Calf':  # Line 3060
+        An_I_req = 0.8 * An_DMIn
+    else:
+        An_I_req = 0.216 * An_BW**0.528 + 0.1 * Trg_MilkProd
+    return An_I_req
+
+
+def calculate_An_I_bal(Dt_IIn: float, An_I_req: float) -> float:
+    """
+    An_I_bal: I balance, mg/d
+    """
+    An_I_bal = Dt_IIn - An_I_req    # Line 3061
+    return An_I_bal
+
+
+### IRON ### 
+def calculate_An_Fe_Clf(Body_Gain: float) -> float:
+    """
+    An_Fe_Clf: Fe requirement for calves, mg/d
+    """
+    An_Fe_Clf = 34 * Body_Gain / 0.25   # Line 3064
+    return An_Fe_Clf
+
+
+def calculate_An_Fe_g(Body_Gain: float) -> float:
+    """
+    An_Fe_g: Fe requirement for growth, mg/d
+    """
+    An_Fe_g = 34 * Body_Gain    # Line 3065
+    return An_Fe_g
+
+
+def calculate_An_Fe_y(An_GestDay: int, An_BW: float) -> float:
+    """
+    An_Fe_y: Fe required for gestation, mg/d
+    """
+    if An_GestDay > 190:    # Line 3066
+        An_Fe_y = 0.025 * An_BW
+    else:
+        An_Fe_y = 0
+    return An_Fe_y
+
+
+def calculate_An_Fe_l(Trg_MilkProd: float) -> float:
+    """
+    An_Fe_l: Fe required for lactation, mg/d
+    """
+    if Trg_MilkProd is None or math.isnan(Trg_MilkProd):    # Line 3067
+       An_Fe_l = 0
+    else:
+        An_Fe_l = 1.0 * Trg_MilkProd
+    return An_Fe_l
+
+
+def calculate_An_Fe_req(An_StatePhys: str, Dt_DMIn_ClfLiq: float, An_Fe_Clf: float, An_Fe_g: float, An_Fe_y: float, An_Fe_l: float) -> float:
+    """
+    An_Fe_req: Fe requirement, mg/d
+    """
+    if An_StatePhys == 'Calf' and Dt_DMIn_ClfLiq > 0:   # Line 3068-3069
+        An_Fe_req = An_Fe_Clf
+    else: 
+        An_Fe_req = An_Fe_g + An_Fe_y + An_Fe_l
```

Comment:
There's a comment in the R code saying: #add An_Fe_m when I move the eqn up here.
Most other _req have the _m - should we include it?